painttiming.bs

```diff
@@ -62,9 +87,34 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+<dfn>Element or pseudo-element</dfn>: A [=document-sourced element=] or a [=generated content pseudo-element=].
+<dfn>Contentful</dfn>: An [=element or pseudo-element=] counts as contentful when one or more of the following apply:
+* The element has a [=text node=] child, with at least one character, excluding [=document white space characters=].
+* The element is a [=replaced image=] representing a [=loaded=] image.
+* The element's [=computed=] style contains a [=loaded=] [=background-image=].
+* The element is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
+* The element is an [=svg element with rendered descendants=].
+* The element's [=computed=] style contains a [=contentful=] and [=paintable=] [=generated content pseudo-element=].
```

Comment:
I don't know what this means. "style containing a pseudo-element" is not really a formally defined concept.

I think we need to use a phrase similar to the one used for define [`pseudo()`](https://drafts.csswg.org/css-pseudo-4/#dom-element-pseudo) (this feature is totally broken due to the total lack of any definition on the lifecycle of CSSPseudoElement but the way it defines which pseudo element gets found is still sound):
> the pseudo-element that would match the selector type with this as its originating element.

So here, I think we need to say something like there is a pseudo element that would match the selector type with the element as the originating element, and the pseudo element is Contentful and paintable.

Now, I think it's better to have a definition on when a pseudo-element is contentfulness & paintable separately from that of element since they're inherently very different kind of objects.

Reply:
hmm yes that would actually make things simpler.